src/Matcher/ArrayMatcher.php

```diff
@@ -179,9 +180,13 @@ private function findNotExistingKeys($pattern, $values)
         $notExistingKeys = array_diff_key($pattern, $values);
 
         return array_filter($notExistingKeys, function ($pattern) use ($values) {
+            if (is_array($pattern)) {
+                return !$this->match($values, $pattern);
```

Comment:
@norzechowicz I'm not pretty sure that we must check if values match to pattern because it already hasn't `array_diff_key($pattern, $values)` and also because OptionalExpander works only with top level match